### PR TITLE
swap ed25519-dalek for ed25519-zebra

### LINKED ISSUE #8055: ed25519-zebra
Are we using the [prehashed](https://docs.rs/ed25519-dalek/1.0.1/ed25519_dalek/index.html?search=prehash) variant of ed25519 anyplace?  If not, then we [should](https://hdevalence.ca/blog/2020-10-04-its-25519am) replace ed25519-dalek with [ed25519-zebra](https://github.com/ZcashFoundation/ed25519-zebra), which better supports batch verification. 

Ain't clear if ed25519-zebra ever wants features not required by zcash, so maybe we should do something else if we use the prehashed variant of ed25519 somewhere.  I once pointed @tomaka to its existence

### PR DESCRIPTION
**Pull request by jakehemmerle**
Please see #8764 as it explains all of this.

Fixes #8055

Polkadot burn in: https://github.com/paritytech/polkadot/pull/5747

### CODE REVIEW

**Comment by jakehemmerle** [comment]
@bkchr fixed all cicd; should be ready

File: primitives/core/Cargo.toml [hunk 1]
```diff
@@ -47,7 +47,7 @@ thiserror = { version = "1.0.30", optional = true }
 bitflags = "1.3"
 
 # full crypto
-ed25519-dalek = { version = "1.0.1", default-features = false, features = ["u64_backend", "alloc"], optional = true }
+ed25519-zebra = { version = "2.2.0", default-features = false, optional = true}
```
Comment (bkchr) [inline primitives/core/Cargo.toml:50]:
```suggestion
ed25519-zebra = { version = "3.0.0", default-features = false, optional = true}
```

**Review by bkchr — APPROVED** [review]

File: primitives/io/src/lib.rs [hunk 2]
```diff
@@ -1941,7 +1944,7 @@ mod tests {
 			// invalid ed25519 signature
```
[the remark this answers is not in the document]
Comment (jakehemmerle) [reply primitives/io/src/lib.rs:1944]: @bkchr yes this comment is incorrect.

while the zero-key behavior is different than dalek, @burdges says its fine. something about 1/16 messages will get accepted or rejected with the zero key anyway. maybe he can elaborate

**Review by davxy — APPROVED** [review]
LGTM.

But please update Cargo.lock first

File: primitives/core/Cargo.toml [hunk 3]
```diff
@@ -47,7 +47,7 @@ thiserror = { version = "1.0.30", optional = true }
 bitflags = "1.3"
 
 # full crypto
-ed25519-zebra = { version = "2.2.0", default-features = false, optional = true}
+ed25519-zebra = { version = "3.0.0", default-features = false, optional = true}
```
[the remark this answers is not in the document]
Comment (jakehemmerle) [reply primitives/core/Cargo.toml:50]: done

**Review by bkchr — APPROVED** [review]

**Review by koute — APPROVED** [review]
From what I can see non-batched performance seems to be unchanged compared to `ed25519-dalek`, so LGTM.

**Review by davxy — APPROVED** [review]

**Pull request merged**